### DISCUSSION TITLE
insert fzf selection into vim

### DISCUSSION THREAD

**Discussion by rbusfield** [junegunn/fzf#3587] [Q&A]
I have a config file (yml-like) that I edit in vim. One section of the config file takes items from a huge list of items (thousands).
I would like to insert the multi-select list from fzf into the current position in vim.
similar to a `:r! cat list.txt | grep -e "foo"`
```
$ cat list.txt
foo
bar
baz
```
What I have tried:
```
$ cat ~/.vimrc
call plug#begin('~/.vim/plugged')
Plug 'junegunn/fzf', { 'do': { -> fzf#install() } }
call plug#end()

$ fzf --version
0.45.0 (tarball)
```
vi file
`:call fzf#run({'source': 'cat list.txt', 'sinklist': 'i'})`
```
Error detected while processing function fzf#run[69]..<SNR>11_callback:
line   23:
Vim(call):E718: Funcref required
Press ENTER or type command to continue
```

**Comment by junegunn** [comment]
```vim
call fzf#run(fzf#wrap({'options': '--multi', 'source': 'cat list.txt', 'sinklist': { list -> append(line('.'), list) }}))
```

* It's always a good idea to use wrap the spec with `fzf#wrap`
* With `--multi`, you can select multiple items using tab and shift-tab
* Use `append` function to append the selected lines under the current line (`line('.')`)
